tests/php/Analyzer/TupleSymbol/FnSymbolTest.php

```diff
@@ -0,0 +1,204 @@
+        ];
+
+        yield 'Start with a number' => [
+            'paramName' => '1-param-3',
+            'error' => true,
+        ];
+
+        yield 'Start with an ampersand' => [
+            'paramName' => '&-param-4',
+            'error' => true,
+        ];
+
+        yield 'Start with a space' => [
+            'paramName' => ' param-5',
+            'error' => true,
+        ];
+    }
+
+    public function testOnlyOneSymbolCanFollowTheAmpersandParameter(): void
+    {
+        $this->expectException(PhelCodeException::class);
+        $this->expectExceptionMessage('Unsupported parameter form, only one symbol can follow the & parameter');
+
+        // This is the same as: (fn [& param-1 param-2])
+        $tuple = Tuple::create(
+            Symbol::create(Symbol::NAME_FN),
+            Tuple::createBracket(
+                Symbol::create('&'),
+                Symbol::create('param-1'),
+                Symbol::create('param-2'),
+            )
+        );
+
+        $this->analyze($tuple);
+    }
+
+    /** @dataProvider providerGetParams */
+    public function testGetParams(Tuple $tuple, array $expectedParams): void
+    {
+        $node = $this->analyze($tuple);
+
+        self::assertEquals($expectedParams, $node->getParams());
+    }
+
+    public function providerGetParams(): Generator
+    {
+        yield '(fn [& param-1])' => [
+            'tuple' => Tuple::create(
+                Symbol::create(Symbol::NAME_FN),
+                Tuple::createBracket(
+                    Symbol::create('&'),
+                    Symbol::create('param-1'),
+                )
+            ),
+            'expectedParams' => [
+                Symbol::create('param-1'),
+            ],
+        ];
+
+        yield '(fn [param-1 param-2 param-3])' => [
+            'tuple' => Tuple::create(
+                Symbol::create(Symbol::NAME_FN),
+                Tuple::createBracket(
+                    Symbol::create('param-1'),
+                    Symbol::create('param-2'),
+                    Symbol::create('param-3'),
+                )
+            ),
+            'expectedParams' => [
+                Symbol::create('param-1'),
+                Symbol::create('param-2'),
+                Symbol::create('param-3'),
+            ],
+        ];
+    }
+
+    /** @dataProvider providerGetBody */
+    public function testGetBody(Tuple $tuple, string $expectedBodyInstanceOf): void
+    {
+        $node = $this->analyze($tuple);
+
+        self::assertInstanceOf($expectedBodyInstanceOf, $node->getBody());
+    }
+
+    public function providerGetBody(): Generator
+    {
+        yield 'DoNode body => (fn [x] x)' => [
+            'tuple' => Tuple::create(
+                Symbol::create(Symbol::NAME_FN),
+                Tuple::createBracket(
+                    Symbol::create('x'),
+                ),
+                Symbol::create('x'),
+            ),
+            'expectedBodyInstanceOf' => DoNode::class,
+        ];
```

Comment:
@jenshaase Could you help me to come write an `tuple` example which will ended up with an  
`expectedBodyInstanceOf => LetNode::class`

Reply:
@jenshaase I tried to write that test case "LetNode example (fn [[x y]] x)", but I got an Exception:
> Phel\Exceptions\PhelCodeException : Can not resolve symbol 'first'

I think the line `Destructure:182` is somehow involved in this...
```php
$accessValue = Tuple::create(
    (Symbol::create('first'))->copyLocationFrom($current),
    $lastListSym
)->copyLocationFrom($current);
```
Any idea why we might get this error? I didn't touch the `Destructure` class logic yet.

Reply:
Oh yes, I forget about this. The `let` expression is also extended and requires a `first` and `next` function (and maybe some others). You have to add them to the `GlobalEnvironment`. I would rewrite the setup function like this:
```
public function setUp(): void
{
  $env = new GlobalEnvironment();
  $env->addDefinition('phel\\core', 'first', new Table()); 
  $env->addDefinition('phel\\core', 'next', new Table());
  $this->analyzer = new Analyzer();
}
```
This code just lets the analyzer know that these functions exists. It doesn't care if the really exists.

Reply:
Done.